kv-storage/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use simperby_common::crypto::*;
+use thiserror::Error;
+
+#[derive(Error, Debug, Serialize, Deserialize, Clone)]
+pub enum Error {
+    /// An unknown error
+    #[error("unknown: {0}")]
+    Unknown(String),
+}
+
+/// TODO: Provide error types.
+///
+/// Note: This trait is quite subject to change.
+#[async_trait]
+pub trait KVStorage {
+    /// Creates an empty storage with the path to newly create.
+    async fn new(path: &str) -> Result<Self, Error>
+    where
+        Self: Sized;
+    /// Open an existing storage with the path given.
+    async fn open(path: &str) -> Result<Self, Error>
+    where
+        Self: Sized;
+    /// Records the current state to the persistent storage.
+    async fn commit_checkpoint(&mut self) -> Result<(), Error>;
+    /// Reverts all the changes made since the last checkpoint.
+    async fn revert_to_latest_checkpoint(&mut self) -> Result<(), Error>;
```

Comment:
Can we commit checkpoint two times in a row and then revert to latest two times?
In other words, will the commits stack up if we call `commit_checkpoint` multiple times?

Reply:
stated.